File: TEQClient/teqclient/src/js/api/microservices/nl-similarity.js
import { sendAsync } from "@/js/utility/request.js";
import { GATEWAY } from "@/js/api/microservices/gateway.js";
import { getRequest } from "@/js/api/api.js";

const API = `${GATEWAY}/nl/similarity/`;

export const findTextSimilarity = (text1, text2) => {
  const request = getRequest("POST", { text1, text2 });
  return sendAsync(API, request);
};

File: TEQServer/mainapp/api/nl_similarity.py
import requests

from mainapp.api.gateway import GATEWAY

URL = f"{GATEWAY}/nl/similarity/"

def calculate_similarity(text1, text2):
    response = requests.post(URL, json={"text1": text1, "text2": text2})

    if response.ok:
        data = response.json()
        if "similarity" not in data:
            raise Exception("No similarity found")
        return data["similarity"]

    raise Exception(f"{response.status_code} - {response.text}")